python/tests/sql_aggregates/test_count_col.py

```diff
@@ -0,0 +1,108 @@
+        pipeline.wait_for_completion(True)
+        out_data = out.to_dict()
+        print(out_data)
+        for datum in expected_data:
+            datum.update({"insert_delete": 1})
+        assert expected_data == out_data
+        pipeline.delete()
+        
+    def add_data(self, new_data, delete: bool = False):
+        key = "delete" if delete else "insert"
+        for datum in new_data:
+            self.data.append({key: datum})
+            
+class Count_Col(TestAggregatesBase):
+    def test_count_col_value(self):
+        pipeline_name = "test_count_col"
+        # validated using postgres
+        expected_data = [{'c1': 4, 'c2': 4, 'c3': 2, 'c4': 4, 'c5': 3, 'c6': 4, 'c7': 3, 'c8': 4}]
+        table_name = "count_col"
+        view_query = f'''CREATE VIEW count_col_view AS SELECT
+                            COUNT(c1) AS c1, COUNT(c2) AS c2, COUNT(c3) AS c3, COUNT(c4) AS c4, COUNT(c5) AS c5, COUNT(c6) AS c6, COUNT(c7) AS c7, COUNT(c8) AS c8
+                        FROM {table_name}'''   
+        self.execute_query(pipeline_name, expected_data, table_name, view_query)
+
+class Count_Col_Groupby(TestAggregatesBase):
+    def test_count_col_groupby(self):
+        pipeline_name = "test_count_col_groupby"
+        # validated using postgres
+        expected_data = [{'id': 0, 'c1': 2, 'c2': 2, 'c3': 1, 'c4': 2, 'c5': 1, 'c6': 2, 'c7': 1, 'c8': 2},
+                         {'id': 1, 'c1': 2, 'c2': 2, 'c3': 1, 'c4': 2, 'c5': 2, 'c6': 2, 'c7': 2, 'c8': 2}]
+        table_name = "count_col_groupby"
+        view_query = f'''CREATE VIEW count_col_view AS SELECT
+                            id, COUNT(c1) AS c1, COUNT(c2) AS c2, COUNT(c3) AS c3, COUNT(c4) AS c4, COUNT(c5) AS c5, COUNT(c6) AS c6, COUNT(c7) AS c7, COUNT(c8) AS c8
+                        FROM {table_name}
+                        GROUP BY id;'''  
+        self.execute_query(pipeline_name, expected_data, table_name, view_query)
+
+class Count_Col_Distinct(TestAggregatesBase):
+    def test_count_col_distinct(self):
+        pipeline_name = "test_count_col_distinct"
+        # validated using postgres
+        expected_data = [{'c1': 2, 'c2': 2, 'c3': 2, 'c4': 4, 'c5': 3, 'c6': 3, 'c7': 3, 'c8': 3}]
+        table_name = "count_col_distinct"
+        view_query = f'''CREATE VIEW count_col_view AS SELECT
+                            COUNT(DISTINCT c1) AS c1, COUNT(DISTINCT c2) AS c2, COUNT(DISTINCT c3) AS c3, COUNT(DISTINCT c4) AS c4, COUNT(DISTINCT c5) AS c5, COUNT(DISTINCT c6) AS c6, COUNT(DISTINCT c7) AS c7, COUNT(DISTINCT c8) AS c8
+                        FROM {table_name}'''
+        self.execute_query(pipeline_name, expected_data, table_name, view_query) 
+
+class Count_Col_Distinct_Groupby(TestAggregatesBase):
+    def test_count_col_distinct_groupby(self):
+        pipeline_name = "test_count_col_distinct_groupby"
+        # validated using postgres
+        expected_data = [{'id': 0, 'c1': 2, 'c2': 1, 'c3': 1, 'c4': 2, 'c5': 1, 'c6': 2, 'c7': 1, 'c8': 2}, 
```

Comment:
need to test this on some inputs that are null and some inputs that produce null results